src/core/utils/Debounce/Debounce.tsx

```diff
@@ -24,7 +24,7 @@ export class Debounce<T extends Object> extends Component<DebounceProps<T>> {
     const { waitFor } = this.props;
     this.timeout = setTimeout(() => {
       callback(value);
-    }, waitFor);
+    }, waitFor) as any;
```

Comment:
Why is this change needed?

Reply:
As discussed, something has changed in typings / typescript / linting. Not sure what exactly, since there were no custom typings for this before the changes of this PR, but nevertheless, typings are broken without the change. For some reason, IDE/TS resolves the ReturnType for setTimeout to be Node.js Timeout rather than a number (browser environment setTimeout). For the setTimeout function however, it is resolved as number => number is not compatible with Node.js Timeout. No idea why this is happening as both declarations are inside the same React class and should resolve to the same environment (browser / Node). There is really no way around it without getting the environment to resolve the ReturnType and setTimeout types the same way.

Reply:
Thanks for clarifying. So be it then, at least for now 👍

Reply:
Yeah, not ideal, but could not find a workaround other than type casting of which the implemented one was the simplest / safest and had the least amount of side effects.